### PR TITLE
User test tweaks and fixing various bugs, also updating copy and questions

### PR DESCRIPTION
**Pull request by hammerfunk**

### CODE REVIEW

File: app/assets/javascripts/application.js.coffee [hunk 1]
```diff
@@ -385,6 +387,15 @@ jQuery ->
       else
         $("#innovative-amount-info").addClass("visuallyhidden")
 
+  # Show trade org fulfilled info when checked yes
+  if $(".question-organisation-fulfill-above-exceptions").size() > 0
+    $(".question-organisation-fulfill-above-exceptions input[type='radio']").bind "propertychange change click keyup input paste", () ->
```
Comment (matthewford) [inline app/assets/javascripts/application.js.coffee]: Line exceeds maximum allowed length<br>Empty parameter list is forbidden

File: app/assets/javascripts/application.js.coffee [hunk 2]
```diff
@@ -385,6 +387,15 @@ jQuery ->
...
+      radio_val = $(".question-organisation-fulfill-above-exceptions input[type='radio']:checked").val()
```
Comment (matthewford) [inline app/assets/javascripts/application.js.coffee]: Line exceeds maximum allowed length

File: app/assets/javascripts/frontend/form-validation.js.coffee [hunk 3]
```diff
@@ -144,8 +144,8 @@ window.FormValidation =
       return
 
     if diffStart < 0 or diffEnd > 0
-      @log_this(question, "validateBetweenDate", "Date should be between #{expDateStart} and #{expDateEnd}.")
-      @addErrorMessage(question, "Date should be between #{expDateStart} and #{expDateEnd}.")
+      @log_this(question, "validateBetweenDate", "Date should be between #{expDateStart} and #{expDateEnd}")
```
Comment (matthewford) [inline app/assets/javascripts/frontend/form-validation.js.coffee]: Line exceeds maximum allowed length

File: app/assets/javascripts/frontend/form-validation.js.coffee [hunk 4]
```diff
@@ -178,8 +205,8 @@ window.FormValidation =
 
     if total_doesnt_match
       question_block = $(".question-block[data-answer='overseas_sales-total-overseas-sales']")
-      @log_this(question, "validateTotalOverseas", "These values should equal the sum of direct and indirect overseas sales.")
-      @appendMessage(question_block, "These values should equal the sum of direct and indirect overseas sales.")
+      @log_this(question, "validateTotalOverseas", "These values should equal the sum of direct and indirect overseas sales")
```
Comment (matthewford) [inline app/assets/javascripts/frontend/form-validation.js.coffee]: Line exceeds maximum allowed length

File: app/assets/javascripts/frontend/form-validation.js.coffee [hunk 5]
```diff
@@ -178,8 +205,8 @@ window.FormValidation =
...
+      @appendMessage(question_block, "These values should equal the sum of direct and indirect overseas sales")
```
Comment (matthewford) [inline app/assets/javascripts/frontend/form-validation.js.coffee]: Line exceeds maximum allowed length

File: app/assets/javascripts/frontend/form-validation.js.coffee [hunk 6]
```diff
@@ -235,8 +262,8 @@ window.FormValidation =
           expDateEnd = dates[1]
 
           if @compareDateInDays(val, expDateStart) < 0 or @compareDateInDays(val, expDateEnd) > 0
-            @log_this(question, "validateDateByYears", "Date should be between #{expDateStart} and #{expDateEnd}.")
-            @appendMessage(q_parent, "Date should be between #{expDateStart} and #{expDateEnd}.")
+            @log_this(question, "validateDateByYears", "Date should be between #{expDateStart} and #{expDateEnd}")
```
Comment (matthewford) [inline app/assets/javascripts/frontend/form-validation.js.coffee]: Line exceeds maximum allowed length

File: app/assets/javascripts/frontend/form-validation.js.coffee [hunk 7]
```diff
@@ -335,6 +362,11 @@ window.FormValidation =
         # console.log "validateTotalOverseas"
         @validateTotalOverseas(question)
 
+      if question.attr("data-answer") == "employees-enter-the-number-of-people-employed-by-your-organisation-in-each-year-of-your-entry" &&
```
Comment (matthewford) [inline app/assets/javascripts/frontend/form-validation.js.coffee]: Line exceeds maximum allowed length

File: app/assets/javascripts/frontend/password-strength-indicator.js [hunk 8]
```diff
@@ -0,0 +1,135 @@
+(function() {
+  "use strict"
```
Comment (matthewford) [inline app/assets/javascripts/frontend/password-strength-indicator.js]: Missing semicolon.

File: app/assets/javascripts/frontend/password-strength-indicator.js [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+  var root = this,
+      $ = root.jQuery;
+
+  if(typeof root.GOVUK === 'undefined') { root.GOVUK = {}; }
```
Comment (matthewford) [inline app/assets/javascripts/frontend/password-strength-indicator.js]: Mixed double and single quotes.

File: app/assets/javascripts/frontend/password-strength-indicator.js [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+
+  var PasswordStrengthIndicator = function(options) {
+    var instance = this;
+
+    $.each([options["password_field"], options["password_confirmation_field"]], function(i, password_field) {
```
Comment (matthewford) [inline app/assets/javascripts/frontend/password-strength-indicator.js]: Line is too long.<br>['password_field'] is better written in dot notation.<br>['password_confirmation_field'] is better written in dot notation.<br>Identifier 'password_field' is not in camel case.

File: app/assets/javascripts/frontend/password-strength-indicator.js [hunk 11]
```diff
@@ -0,0 +1,135 @@
...
+      var update = function() {
+        var password = $(options["password_field"]).val(),
```
Comment (matthewford) [inline app/assets/javascripts/frontend/password-strength-indicator.js]: ['password_field'] is better written in dot notation.

File: app/assets/javascripts/frontend/password-strength-indicator.js [hunk 12]
```diff
@@ -0,0 +1,135 @@
...
+            passwordConfirmation = $(options["password_confirmation_field"]).val();
```
Comment (matthewford) [inline app/assets/javascripts/frontend/password-strength-indicator.js]: ['password_confirmation_field'] is better written in dot notation.

File: app/assets/javascripts/frontend/password-strength-indicator.js [hunk 13]
```diff
@@ -0,0 +1,135 @@
...
+        instance.updateIndicator(password, passwordConfirmation, options);
+      };
+      $(password_field).debounce("keyup", update, 50);
```
Comment (matthewford) [inline app/assets/javascripts/frontend/password-strength-indicator.js]: Identifier 'password_field' is not in camel case.

File: app/assets/javascripts/frontend/password-strength-indicator.js [hunk 14]
```diff
@@ -0,0 +1,135 @@
...
+    });
+
+    $(options["password_strength_guidance"]).attr("aria-live", "polite").attr("aria-atomic", "true");
```
Comment (matthewford) [inline app/assets/javascripts/frontend/password-strength-indicator.js]: Line is too long.<br>['password_strength_guidance'] is better written in dot notation.

File: app/assets/javascripts/frontend/password-strength-indicator.js [hunk 15]
```diff
@@ -0,0 +1,135 @@
...
+  };
+
+  PasswordStrengthIndicator.prototype.updateIndicator = function(password, passwordConfirmation, options) {
```
Comment (matthewford) [inline app/assets/javascripts/frontend/password-strength-indicator.js]: Line is too long.

File: app/assets/javascripts/frontend/password-strength-indicator.js [hunk 16]
```diff
@@ -0,0 +1,135 @@
...
+    var guidance = [];
+
+    var result = zxcvbn(password, options["weak_words"]);
```
Comment (matthewford) [inline app/assets/javascripts/frontend/password-strength-indicator.js]: ['weak_words'] is better written in dot notation.

File: app/assets/javascripts/frontend/password-strength-indicator.js [hunk 17]
```diff
@@ -0,0 +1,135 @@
...
+    if (password.length > 0) {
+      if (options["min_password_length"] && password.length < parseInt(options["min_password_length"])) {
```
Comment (matthewford) [inline app/assets/javascripts/frontend/password-strength-indicator.js]: Line is too long.<br>['min_password_length'] is better written in dot notation.<br>Missing radix parameter.

File: app/assets/javascripts/frontend/password-strength-indicator.js [hunk 18]
```diff
@@ -0,0 +1,135 @@
...
+        guidance.push('password-too-short');
```
Comment (matthewford) [inline app/assets/javascripts/frontend/password-strength-indicator.js]: Mixed double and single quotes.

File: app/assets/javascripts/frontend/password-strength-indicator.js [hunk 19]
```diff
@@ -0,0 +1,135 @@
...
+      }
+
+      var isPasswordNotStrongEnough = (result.score < options["strong_passphrase_boundary"]);
```
Comment (matthewford) [inline app/assets/javascripts/frontend/password-strength-indicator.js]: ['strong_passphrase_boundary'] is better written in dot notation.

File: app/assets/javascripts/frontend/password-strength-indicator.js [hunk 20]
```diff
@@ -0,0 +1,135 @@
...
+
+      var aWeakWordFoundInPassword = $(options["weak_words"]).is(function(i, weak_word) {
```
Comment (matthewford) [inline app/assets/javascripts/frontend/password-strength-indicator.js]: ['weak_words'] is better written in dot notation.<br>Identifier 'weak_word' is not in camel case.

File: app/assets/javascripts/application.js.coffee [hunk 21]
```diff
@@ -385,6 +387,16 @@ jQuery ->
       else
         $("#innovative-amount-info").addClass("visuallyhidden")
 
+  # Show trade org fulfilled info when checked yes
+  trade_org_q = ".question-organisation-fulfill-above-exceptions"
+  if $(trade_org_q).size() > 0
+    $("#{trade_org_q} input[type='radio']").bind "propertychange change click keyup input paste", () ->
```
Comment (matthewford) [inline app/assets/javascripts/application.js.coffee]: Line exceeds maximum allowed length<br>Empty parameter list is forbidden

File: app/assets/javascripts/frontend/password-strength-indicator.js [hunk 22]
```diff
@@ -0,0 +1,135 @@
...
+        return (password.indexOf(weak_word) >= 0);
```
Comment (matthewford) [inline app/assets/javascripts/frontend/password-strength-indicator.js]: Identifier 'weak_word' is not in camel case.

File: app/assets/javascripts/frontend/password-strength-indicator.js [hunk 23]
```diff
@@ -0,0 +1,135 @@
...
+      });
+      if (isPasswordNotStrongEnough && aWeakWordFoundInPassword) {
+        guidance.push('parts-of-email');
```
Comment (matthewford) [inline app/assets/javascripts/frontend/password-strength-indicator.js]: Mixed double and single quotes.

**Comment by RuslanHamidullin** [comment]
:bomb:

**Pull request merged**